libsel4/include/interfaces/sel4.xml

```diff
@@ -372,6 +372,13 @@
                 description="Capability to the scheduling context that the TCB should run on. If the scheduling context is already bound to a notification or TCB that is not this TCB this operation will fail. Similarly, if this TCB is already bound to a scheduling context that is not this scheduling context, this will also fail."/>
             <param dir="in" name="fault_ep" type="seL4_CPtr"
                 description="CPTR to the endpoint which receives IPCs when this thread faults."/>
+            <param dir="in" name="fault_ep_data" type="seL4_Word"
+                description="Optionally set the badge of the new fault handler Endpoint. If both this and fault_ep_rights are set to zero, then this parameter has no effect."/>
```

Comment:
Thank you for the reminder, I forgot to talk about this design decision. The intention is to both support not modifying the fault handler to maintain compatibility with existing code, and also allow for setting the badge to `0`, which is a valid and I think fairly common value.

Since setting the rights to `0` (i.e. `seL4_NoRights`) would result in an invalid fault handler, I used the combination of both being `0` as a special case to flag not wanting to modify the fault handler cap. With this approach, the badge can still be set to `0` with a combination of `badge=0` and `rights=seL4_AllRights`.

Writing this out has made me realise that an alternative approach could be to just use the rights parameter for this purpose, instead of requiring that both are set to `0`. However, while this would slightly simplify the API, I could see it causing confusion if someone doesn't consider it and tries to set a badge to a specific value without changing the rights parameter away from `0`.

Reply:
@corlewis, could you use the fact that every invocation has a length provided by the seL4_MessageInfo_t which you can use to have unambiguously unused parameters by having encodings where the parameters are unused? This is what I believe @Indanz is meaning when talking about optional parameters.

Reply:
> With this approach, the badge can still be set to 0 with a combination of badge=0 and rights=seL4_AllRights

Only unbadged caps can be rebadged, so the badge value is already 0. In effect the above is a specific corner case of only changing the rights without changing the badge value.

Reply:
Right, I explicitly looked at `updateCapData` as part of this work but I then forgot that only unbadged caps could be rebadged. Considering this, I agree with you that the parameters should be independent.